src/csharp/Microsoft.Spark/SparkContext.cs

```diff
@@ -20,6 +20,10 @@ namespace Microsoft.Spark
     /// </summary>
     public sealed class SparkContext : IJvmObjectReferenceProvider
     {
+        public string TempDir { get; set; }
```

Comment:
Why is this public? Note that the APIs should be comparable to the Scala side. Same for `EncryptionEnabled`

Reply:
I need to access these properties from `Broadcast` , they are not exposed in Scala's SparkContext, but in Pyspark's they are public. I can make them public with private setters. Please let me know the best approach, thanks!

Reply:
Do you need this somewhere else? If not, why not create the temp dir when you need it? We don't need to follow PySpark line by line.

Reply:
Yes I'll do that, thanks!